src/main/java/net/tridentsdk/chat/ChatColor.java

```
@@ -0,0 +1,192 @@
+    DARK_GRAY('8'),
+
+    /**
+     * Blue, §9
+     */
+    BLUE('9'),
+
+    /**
+     * Green, §a
+     */
+    GREEN('a'),
+
+    /**
+     * Aqua, §b
+     */
+    AQUA('b'),
+
+    /**
+     * Red, §c
+     */
+    RED('c'),
+
+    /**
+     * Light Purple, §d
+     */
+    LIGHT_PURPLE('d'),
+
+    /**
+     * Yellow, §e
+     */
+    YELLOW('e'),
+
+    /**
+     * White, §f
+     */
+    WHITE('f'),
+
+    /**
+     * Obfuscated, §k
+     */
+    OBFUSCATED('k'),
+
+    /**
+     * Bold, §l
+     */
+    BOLD('l'),
+
+    /**
+     * Strikethrough, §m
+     */
+    STRIKETHROUGH('m'),
+
+    /**
+     * Underline, §n
+     */
+    UNDERLINE('n'),
+
+    /**
+     * Italic, §o
+     */
+    ITALIC('o'),
+
+    /**
+     * Reset, §r
+     */
+    RESET('r');
+
+    @Getter
+    private char colorChar;
+
+    private ChatColor(char colorChar) {
+        this.colorChar = colorChar;
+    }
+
+    /**
+     * Gets if this chat color is a format color.
+     *
+     * @return True iff it is.
+     */
+    public boolean isFormat() {
+        return 'k' <= colorChar && colorChar <= 'r';
+    }
+
+    /**
+     * Gets if this chat color is a color.
+     *
+     * @return The color.
+     */
+    public boolean isColor() {
+        return !isFormat();
+    }
+
+    /**
+     * Gets a string representation of this color, in the form §{@code x}, where x is the color's character.
+     *
+     * @return The color.
+     */
+    public String toString() {
+        return "\u00A7" + colorChar;
+    }
+
+    private static Map<Character, ChatColor> charToColor = new HashMap<Character, ChatColor>() {
+        {
+            for (ChatColor c : ChatColor.values()) {
+                put(c.colorChar, c);
+            }
+        }
+    };
+
+    /**
+     * Gets a chat color from a given character.
+     *
+     * @param colorChar The color's character.
+     *
+     * @return The color, or null if not found.
+     */
+    public static ChatColor getColor(char colorChar) {
```

Please throw an exception rather than returning if the chat color is not found.